modules/likes/jetpack-likes-settings.php

```diff
@@ -599,13 +599,13 @@ function admin_settings_callback() {
 
 		// Checked (enabled)
 		switch( $new_state ) {
-			case 'off' :
+			case 'false':
```

Comment:
Here you're changing the value from one string to another string, but the intention was to change that value to a boolean. Same with all the other values down there. Let's use `true` and `false` instead.

Plus, more importantly, on line 593 and 596 there's a default setting that still sets the variables to `on` if the value is not defined.